File: Python/num2.py
#!/usr/bin/env python3
# Programa: Manipulacao de Números - Capitulo 2 (Python)
# Ambiente: Ubuntu / vscode
import sys, random
try:
    argv1 = float(sys.argv[1])
except (IndexError, ValueError):
    argv1 = None

N1 = random.randint(1,9)
n1 = None

print("LITERAL NUMERICA:")
print("\033[7m41\033[0m")
 
print("\nVALOR INICIAL DA VARIAVEL:")
print("n1 = ", n1)
 
n1 = 7
print("\nNOVA ATRIBUICAO:")
print("n1 = ", n1)
 
print("\nENTRADA VIA ARGUMENTO:")
n1 = argv1
print("n1 = ", n1)

print("\nENTRADA INTERATIVA:")
x = input("Digite um valor inteiro nao negativo: ")
try:
    n1 = float(x)
except ValueError:
    n1 = 0
print("n1 = ", n1)
  
print("\nOPERACOES ARITMETICAS:")
print("n1 + N1 = ", n1 + N1)
print("n1 - N1 = ", n1 - N1)
print("n1 * N1 = ", n1 * N1)
print("n1 / N1 = ", n1 / N1)
print("n1 // N1 = ", n1 // N1)
print("n1 % N1 = ", n1 % N1)
print("n1 ** N1 = ", n1 ** N1)
  	
print("\nDESAFIO:")
x = input("Adivinhe o valor da constante N1 (1-9): ")
palpite = int(x) if x.isdigit() else 0
print(f"Parabéns, você acertou! Era {N1}") if palpite == N1 else print(f"Não foi desta vez! Era {N1}")

File: C/src/cap2/num2.c
/*
Programa: Manipulacao de Números - Capitulo 2 (C)
Ambiente: Ubuntu - gcc / nano
*/

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *argv[]){
    int n1, palpite;
    srand(time(NULL));                  // inicializa o gerador com o tempo atual
    const int N1 = (rand() % 9) + 1;    /* número aleatório de 1 a 9 */
    
    printf("LITERAL NUMERICA:\n");
    printf("%d\n", 41);
        
    printf("\nDECLARACAO DE VARIAVEL -> VALOR INDEFINIDO:\n");
    printf("n1 = %d\n", n1);
     
    n1 = 7;
    printf("\nINICIALIZACAO:\n");
    printf("n1 = %d\n", n1);
    
    n1 = 10;
    printf("\nNOVA ATRIBUICAO:\n");
    printf("n1 = %d\n", n1);
    
    printf("\nENTRADA VIA ARGUMENTO:\n");
    printf("n1 = %d\n", (argc > 1) ? atoi(argv[1]) : 0);


    printf("\nENTRADA INTERATIVA:\n");
    printf("Digite um valor: ");
    scanf("%d", &n1);
    printf("n1 = %d\n", n1);
    
    printf("\nOPERACOES ARITMETICAS:\n");
    printf("n1 + N1 = %d\n", n1 + N1);
    printf("n1 - N1 = %d\n", n1 - N1);
    printf("n1 * N1 = %d\n", n1 * N1);
    printf("n1 / N1 = %d\n", n1 / N1);
    printf("n1 %% N1 = %d\n", n1 % N1);

    printf("\nDESAFIO:\n");
    printf("Adivinhe o valor da constante N1 (1-9): ");
    scanf("%d", &palpite);
    printf("%s N1 = %d.\n", (palpite == N1) ? "Acertou." : "Nao foi dessa vez.", N1);

    return 0;
}
